src/smart-components/role/add-role-new/add-permissions.js

```diff
@@ -80,13 +99,20 @@ const AddPermissionsTable = ({ selectedPermissions, setSelectedPermissions, ...p
         setPermissions(accessWrapper(access, filters));
     }, [ filters ]);
 
+    useEffect(() => {
+        const resolvedPermissions = resolveSplats(selectedPermissions, permissions.data);
```

Comment:
No need to assign to a constant, also could be better to use state callback so we are in sync
```JS
setSelectedPermissions(() => resolveSplats(selectedPermissions, permissions.data));
```

Reply:
Looking at the second useEffect, it is better to combine effects together so we don't fire unwanted re-renders.
```JS
useEffect(() => {
  if (baseRole) {
    setSelectedPermissions(() => resolveSplats(
      baseRole.access?.map(({ permission  }) => ({ uuid: permission })),
      permissions.data
    ));
  } else {
    setSelectedPermissions(() => resolveSplats(selectedPermissions, permissions.data));
  }
}, [ permissions.data, baseRole ])
```

Here you can actually see that if there is any base role selected it will set selected permissions based on this and if not set it will always use the other function. With the previous implementation you would have to keep in mind that the selected permissions are being set from other effect and check which one takes precedens.